### ISSUE TITLE
Stop squelching error messages in CustomPouchErrors

### ISSUE THREAD

**Issue by nolanlawson** [pouchdb/pouchdb#5214]
#5213 reveals that our custom errors frequently squelch the underlying error message, leading to issues like #4803, #5104, and #4670 (which I've combined into this new issue).

Once #5213 is merged and we have solid unit tests in place, we can go back and make sure that the error message is more useful. I'd also like `generateErrorFromResponse()` to handle cases where the incoming `res` object is not JSON (running the tests, I saw both strings and `Buffers` (!)). If an HTML page is returned with an error message, then we should probably just make that whole string into the resopnse. Ditto for `Buffer`s.

Here's an example of the kind of squelching behavior that I think can be improved:

```js
var response = { "error": "bad_request", "reason": "Referer header required." };
var err = generateErrorFromResponse(response);
err.status.should.equal(400);
err.name.should.equal("bad_request");
err.message.should.equal("Something wrong with the request");
```

Or this one: 

```js
var response = "<!doctype html><html>YO THIS IS AN ERROR</html>";
var err = generateErrorFromResponse(response);
err.status.should.equal(500);
err.name.should.equal("unknown_error");
err.message.should.equal("Database encountered an unknown error");
```

**Comment by nolanlawson** [comment]
For the test case, see `test.errors.js` in https://github.com/pouchdb/pouchdb/pull/5213.

**Comment by daleharvey** [comment]
Fixed in https://github.com/pouchdb/pouchdb/commit/d31bf170ccfa54ca58e6afd43fe9f496e639e1a9
